addons/web/static/tests/views/fields/numeric_fields_tests.js

```diff
@@ -306,4 +310,64 @@ QUnit.module("Fields", (hooks) => {
             });
         }
     );
+
+    QUnit.test(
+        "useNumpadDecimal should synchronize handlers on input elements",
+        async function (assert) {
+            target = getFixture();
+            patchWithCleanup(localization, { decimalPoint: ",", thousandsSep: "." });
+
+            /**
+             * Takes an array of input elements and asserts that each has the correct event listener.
+             * @param {HTMLInputElement[]} inputEls
+             */
+            function testInputElements(inputEls) {
+                for (const inputEl of inputEls) {
+                    inputEl.focus();
+                    const numpadDecimalEvent = new KeyboardEvent("keydown", {
+                        code: "NumpadDecimal",
+                        key: ".",
+                    });
+                    numpadDecimalEvent.preventDefault = () => assert.step("preventDefault");
+                    inputEl.dispatchEvent(numpadDecimalEvent);
+
+                    // dispatch an extra keydown event and assert that it's not default prevented
+                    const extraEvent = new KeyboardEvent("keydown", { code: "Digit1", key: "1" });
+                    extraEvent.preventDefault = () => {
+                        throw new Error("should not be default prevented");
+                    };
+                    inputEl.dispatchEvent(extraEvent);
+                    assert.verifySteps(["preventDefault"]);
+                }
+            }
+
+            class MyComponent extends Component {
+                setup() {
+                    useNumpadDecimal();
+                    this.state = useState({ showOtherInput: false });
+                }
+            }
+            MyComponent.template = xml`
+                <main t-ref="numpadDecimal">
+                    <input type="text" placeholder="input 1" />
+                    <input t-if="state.showOtherInput" type="text" placeholder="input 2" />
+                </main>
+            `;
+            const comp = await mount(MyComponent, target, { env: await makeTestEnv() });
+
+            // Initially, only one input should be rendered.
+            let inputs = target.querySelectorAll("main > input");
+            assert.strictEqual(inputs.length, 1);
+            testInputElements(inputs);
```

Comment:
you can also do this, but that's up to you ;-)
```suggestion
            assert.containsOnce(target, "main > input");
            testInputElements(target.querySelectorAll("main > input");
```

(same below with `assert.containsN(..., 2)`)